src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -82,18 +84,24 @@ private TypeWithState(TypeSymbol type, NullableFlowState state)
 
         public TypeWithState WithSuppression(bool suppress) => suppress ? new TypeWithState(Type, NullableFlowState.NotNull) : this;
 
-        public TypeWithAnnotations ToTypeWithAnnotations()
+        public TypeWithAnnotations ToTypeWithAnnotations(CSharpCompilation compilation, bool asAnnotatedType = false)
```

Comment:
Hard to understand the value of `asAnnotatedType` given that in several cases we just ignore it and always return an annotated type.